observable_operator.go

```diff
@@ -529,16 +531,37 @@ func (o *ObservableImpl) BufferWithTimeOrCount(timespan Duration, count int, opt
 		send := make(chan struct{})
 		mutex := sync.Mutex{}
 
-		checkBuffer := func() {
+		// checkBuffer will send buffered units of at most size count, unless
+		// flush is true.
+		checkBuffer := func(flush bool) {
 			mutex.Lock()
-			if len(buffer) != 0 {
-				if !Of(buffer).SendContext(ctx, next) {
-					mutex.Unlock()
-					return
+			defer mutex.Unlock()
+
+			length := len(buffer)
+			if length != 0 {
+				var last int
+				defer func() {
+					// create a copy of buffer, less whatever was already sent
+					t := make([]interface{}, length-(last+1))
+					copy(t, buffer[last+1:length])
```

Comment:
What's the rationale for a copy instead of a slicing operation?

Reply:
I wanted to release the original buffer as soon as possible.  If a slice is made, the original buffer could be held until the next "timeout".

Reply:
Makes sense